### ISSUE TITLE
Missing dist/quasar.esm.prod.js with quasar v2.8.0

### ISSUE THREAD

**Issue by rpastro** [quasarframework/quasar#14391]
### What happened?

The dist/quasar.esm.prod.js file seems to be missing after upgrading to quasar v2.8.0.  The package.json still lists this file as module so I'm assuming it should be there.

```
"module": "dist/quasar.esm.prod.js"
```

### What did you expect to happen?

The dist/quasar.esm.prod.js file should exist with quasar v2.8.0.

### Reproduction URL

NA

### How to reproduce?

npm install quasar@2.8.0

quasar.esm.prod.js is not in node_modules/quasar/dist folder.

### Flavour

Quasar CLI with Webpack (@quasar/cli | @quasar/app-webpack)

### Areas

Quasar CLI Commands/Configuration (@quasar/cli | @quasar/app-webpack | @quasar/app-vite)

**Comment by rstoenescu** [comment]
Hi @rpastro ,

That dist file was not being used at all. However, listing it in package.json is indeed not correct and will fix it.
For the sake of it, please confirm that you are reporting just this inconsistency and your compiling is not affected (this is my expectation anyways).

**Comment by rpastro** [comment]
@rstoenescu   Thank you for the quick feedback.   Actually, our project is referencing this file for unit tests.  I don't recall the history behind that, but I believe it was needed because the unit tests are written using ESM in Node.

I noticed this error because our unit tests started failing after I upgraded to quasar v2.8.0.

Below is the current jest.config.js file used in our project. You may see that we are currently pointing to this file under moduleName Wrapper (`'^quasar$': 'quasar/dist/quasar.esm.prod.js'`).

```
'use strict';
const esModules = ['quasar', 'quasar/lang', 'lodash-es'].join('|');

module.exports = {
  collectCoverage: true,
  collectCoverageFrom: [
    '<rootDir>/src/**/*.vue',
    '<rootDir>/src/**/*.js'
  ],
  coverageDirectory: '<rootDir>/test/coverage',
  coveragePathIgnorePatterns: [
    '/node_modules/',
    '.d.ts$'
  ],
  coverageReporters: [
    'json',
    'lcov',
    'text-summary'
  ],
  globals: {
    __DEV__: true,
    // TODO: Remove if resolved natively
    // See https://github.com/vuejs/vue-jest/issues/175
    'vue-jest': {
      pug: { doctype: 'html' }
    },
    // Remove if using `const enums`
    // See https://huafu.github.io/ts-jest/user/config/isolatedModules#example
    'ts-jest': {
      isolatedModules: true
    }
  },
  moduleFileExtensions: [
    'vue',
    'js'
  ],
  moduleNameMapper: {
    '^quasar$': 'quasar/dist/quasar.esm.prod.js',
    '@/(.*)$': '<rootDir>/src/$1',
    '^~/(.*)$': '<rootDir>/$1',
    '^src/(.*)$': '<rootDir>/src/$1',
    '^app/(.*)$': '<rootDir>/$1',
    '^components/(.*)$': '<rootDir>/src/components/$1',
    '^layouts/(.*)$': '<rootDir>/src/layouts/$1',
    '^pages/(.*)$': '<rootDir>/src/pages/$1',
    '^assets/(.*)$': '<rootDir>/src/assets/$1',
    '^boot/(.*)$': '<rootDir>/src/boot/$1',
    '.*css$': '@quasar/quasar-app-extension-testing-unit-jest/stub.css'
  },
  reporters: [
    'default'
  ],
  setupFilesAfterEnv: [
    './test/jestSetup.js'
  ],
  snapshotSerializers: [
    'jest-serializer-vue'
  ],
  testMatch: [
    '<rootDir>/test/*Spec.js',
    '<rootDir>/test/**/*Spec.js'
  ],
  transform: {
    '.*\\.vue$': 'vue-jest',
    '.*\\.js$': 'babel-jest',
    '.+\\.(css|styl|less|sass|scss|svg|png|jpg|ttf|woff|woff2)$': 'jest-transform-stub'
  },
  transformIgnorePatterns: [`node_modules/(?!(${esModules}))`],
  verbose: true
};
```

**Comment by rstoenescu** [comment]
Can you point it to quasar.esm.js instead and report back if you have any issues pls?

**Comment by DavideSegullo** [comment]
> Hi @rpastro ,
> 
> That dist file was not being used at all. However, listing it in package.json is indeed not correct and will fix it. For the sake of it, please confirm that you are reporting just this inconsistency and your compiling is not affected (this is my expectation anyways).

Hello,
actually vite for example use it, yesterday I could not install quasar along with vite

**Comment by rstoenescu** [comment]
@DavideSegullo nope, it does not. Use `@quasar/vite-plugin` when on a Vite project (referring to the non Quasar CLI with Vite).

**Comment by rpastro** [comment]
@rstoenescu   Unfortunately I get the following errors if I point to `quasar.esm.js`.  I apologize, but it is 3 AM in east coast so I didn't try to debug it. :) 

```
FAIL test/settingsSpec.js
  ● Test suite failed to run

    ReferenceError: __QUASAR_SSR_CLIENT__ is not defined

      1 | import { describe, it } from '@jest/globals';
    > 2 | import { installQuasarPlugin, qLayoutInjections } from '@quasar/quasar-app-extension-testing-unit-jest';
        | ^
      3 | import { mount } from '@vue/test-utils';
      4 | import Vuex from 'vuex';
      5 | import { nextTick } from 'vue';

      at Object.<anonymous> (node_modules/quasar/dist/quasar.esm.js:34:5)
      at Object.<anonymous> (node_modules/@quasar/quasar-app-extension-testing-unit-jest/src/helpers/install-quasar-plugin.ts:4:1)
      at Object.<anonymous> (node_modules/@quasar/quasar-app-extension-testing-unit-jest/src/helpers/index.ts:1:1)
      at Object.<anonymous> (test/settingsSpec.js:2:1)

FAIL test/loginSpec.js
  ● Test suite failed to run

    ReferenceError: __QUASAR_SSR_CLIENT__ is not defined

      1 | import { describe, it } from '@jest/globals';
    > 2 | import { installQuasarPlugin } from '@quasar/quasar-app-extension-testing-unit-jest';
        | ^
      3 | import { mount } from '@vue/test-utils';
      4 | import { QBtn } from 'quasar';
      5 | import Vuex from 'vuex';

      at Object.<anonymous> (node_modules/quasar/dist/quasar.esm.js:34:5)
      at Object.<anonymous> (node_modules/@quasar/quasar-app-extension-testing-unit-jest/src/helpers/install-quasar-plugin.ts:4:1)
      at Object.<anonymous> (node_modules/@quasar/quasar-app-extension-testing-unit-jest/src/helpers/index.ts:1:1)
      at Object.<anonymous> (test/loginSpec.js:2:1)
```

**Comment by DavideSegullo** [comment]
> @DavideSegullo nope, it does not. Use `@quasar/vite-plugin` when on a Vite project (referring to the non Quasar CLI with Vite).

I used vite-plugin, but i got errors when i had to import quasar into main.ts, that's the step: https://quasar.dev/start/vite-plugin#using-quasar.

Something like:
```bash
[plugin vite:dep-scan] Failed to resolve entry for package 'quasar'
```

**Comment by rstoenescu** [comment]
Decided to add back quasar.esm.prod.js and released Quasar v2.8.1.

**Comment by rpastro** [comment]
@rstoenescu   Thank you very much for such a quick turnaround.   I just updated to v2.8.1, but now I'm getting other errors when running the unit tests.  I don't know if I have to change something on our side or if something else got broken in the upgrade to v2.8.x. 

```
> npx cross-env NODE_OPTIONS='--experimental-vm-modules' npx jest --config ./jest.config.js --coverage --detectOpenHandles --no-color

(node:30072) ExperimentalWarning: VM Modules is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
PASS test/settingsSpec.js (10.874 s)
  Settings.vue
    ✓ Settings - Change Language Success (53 ms)
    ✓ Settings - Change Same Language (16 ms)
    ✓ Settings - Change Language Error (11 ms)

  console.warn
    [Vue warn]: Failed to resolve component: q-toolbar
    If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement. 
      at <NglsSettings ref="VTU_COMPONENT" > 
      at <VTUROOT>



      at warn (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:40:17)
      at resolveAsset (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:2847:13)
      at resolveComponent (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:2796:12)
      at Proxy.render (src/pages/Settings.vue:79:32)
      at renderComponentRoot (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:885:44)
      at ReactiveEffect.componentUpdateFn [as fn] (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:5532:57)
      at ReactiveEffect.run (node_modules/@vue/reactivity/dist/reactivity.cjs.js:189:25)

  console.warn
    [Vue warn]: Failed to resolve component: q-icon
    If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement. 
      at <NglsSettings ref="VTU_COMPONENT" > 
      at <VTUROOT>



      at warn (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:40:17)
      at resolveAsset (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:2847:13)
      at resolveComponent (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:2796:12)
      at Proxy.render (src/pages/Settings.vue:81:29)
      at renderComponentRoot (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:885:44)
      at ReactiveEffect.componentUpdateFn [as fn] (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:5532:57)
      at ReactiveEffect.run (node_modules/@vue/reactivity/dist/reactivity.cjs.js:189:25)

  console.warn
    [Vue warn]: Failed to resolve component: q-select
    If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement. 
      at <NglsSettings ref="VTU_COMPONENT" > 
      at <VTUROOT>



      at warn (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:40:17)
      at resolveAsset (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:2847:13)
      at resolveComponent (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:2796:12)
      at Proxy.render (src/pages/Settings.vue:83:31)
      at renderComponentRoot (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:885:44)
      at ReactiveEffect.componentUpdateFn [as fn] (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:5532:57)
      at ReactiveEffect.run (node_modules/@vue/reactivity/dist/reactivity.cjs.js:189:25)


  console.warn
    [Vue warn]: Failed to resolve component: q-page
    If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement. 
      at <NglsSettings ref="VTU_COMPONENT" > 
      at <VTUROOT>



      at warn (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:40:17)
      at resolveAsset (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:2847:13)
      at resolveComponent (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:2796:12)
      at Proxy.render (src/pages/Settings.vue:85:29)
      at renderComponentRoot (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:885:44)
      at ReactiveEffect.componentUpdateFn [as fn] (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:5532:57)
      at ReactiveEffect.run (node_modules/@vue/reactivity/dist/reactivity.cjs.js:189:25)

... and more
```

**Comment by rstoenescu** [comment]
@rpastro Forgot to auto-register all components and directives for that entry-point. Released v2.8.2. Thanks for pointing this out!

**Comment by donnyv** [comment]
Thanks for adding back in. Without this issue I would of never known that Quasar has a ESM build. I couldn't find anything about it in your docs.
